hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/AbstractRolloutExecutor.java

```diff
@@ -0,0 +1,236 @@
+
+        final JpaRollout jpaRollout = (JpaRollout) rollout;
+
+        switch (rollout.getStatus()) {
+        case CREATING:
+            countingCreationRetries(jpaRollout, this::handleCreateRollout, (retries, e) -> {
+                LOGGER.warn("Exceeded maxRetries[{}] when trying to create rollout with id [{}] and name '{}' due to: ",
+                        retries, rollout.getId(), rollout.getName(), e);
+                handleErrorCreating(jpaRollout);
+            });
+            break;
+        case DELETING:
+            countingDeletingRetries(jpaRollout, this::handleDeleteRollout, (retries, e) -> {
+                LOGGER.warn("Exceeded maxRetries[{}] when trying to delete rollout with id [{}] and name '{}' due to: ",
+                        retries, rollout.getId(), rollout.getName(), e);
+                handleErrorDeleting(jpaRollout);
+            });
+            break;
+        case READY:
+            handleReadyRollout(jpaRollout);
+            break;
+        case STARTING:
+            countingStartingRetries(jpaRollout, this::handleStartingRollout, (retries, e) -> {
+                LOGGER.warn("Exceeded maxRetries[{}] when trying to start rollout with id [{}] and name '{}' due to: ",
+                        retries, rollout.getId(), rollout.getName(), e);
+                handleErrorStarting(jpaRollout);
+            });
+            break;
+        case RUNNING:
+            handleRunningRollout(jpaRollout);
+            break;
+        case STOPPING:
+            handleStopRollout(jpaRollout);
+            break;
+        default:
+            LOGGER.error("Rollout in status {} not supposed to be handled!", rollout.getStatus());
+            break;
+        }
+    }
+
+    protected abstract void handleCreateRollout(final JpaRollout rollout);
+
+    protected abstract void handleDeleteRollout(final JpaRollout rollout);
+
+    protected abstract void handleReadyRollout(final JpaRollout rollout);
+
+    protected abstract void handleStartingRollout(final JpaRollout rollout);
+
+    protected abstract void handleRunningRollout(final JpaRollout rollout);
+
+    protected abstract void handleStopRollout(final JpaRollout rollout);
+
+    protected void handleErrorCreating(final JpaRollout rollout) {
+        setRolloutErrorState(rollout);
+    }
+
+    protected void handleErrorStarting(final JpaRollout rollout) {
+        setRolloutErrorState(rollout);
+    }
+
+    protected void handleErrorDeleting(final JpaRollout rollout) {
+        setRolloutErrorState(rollout);
+    }
+
+    private void setRolloutErrorState(final JpaRollout rollout) {
+        if (!cleanupScheduledActions(rollout)) {
+            // could not clean up scheduled actions
+            LOGGER.warn("Could not clean up scheduled actions for rollout with id {}", rollout.getId());
+        }
+        setErrorStateForRolloutGroups(rollout);
+        rollout.setStatus(Rollout.RolloutStatus.ERROR_STARTING);
```

Comment:
method is used in all three cases: creating, starting and deleting, however status is always "Error_Starting", please adapt accordingly

Reply:
done